### ISSUE TITLE
Unable to run on M1 due to failed provider installation

### ISSUE THREAD

**Issue by pecigonzalo** [hashicorp/terraform-cdk#909]
### Community Note

- Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
- Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
- If you are interested in working on this issue or have submitted a pull request, please leave a comment

### cdktf & Language Versions

Any lang
```
❯ cdktf --version
0.5.0
```

### Affected Resource(s)

`cdktf-cli`

### Debug Output

https://gist.github.com/pecigonzalo/80b586232f124bcf63eee2e5911f7cae

### Expected Behavior

We should be able to override the arch for the providers, either via config, by using `arch -x86_64` or some other way.

### Actual Behavior

Even when running an x86_64 version of `node` for the CDK and/or running the CDK with `arch -x86_64 cdktf get`, many providers fail to download because they are not available for m1.

### Steps to Reproduce

- Install `cdktf` using `homebrew` or `npx cdktf` (npx must be x86_64)
- Initialize a new project in any lang with `cdktf init`
- Add a provider like `"terraformProviders": ["terraform-providers/docker@~> 2.0"]`
- Try to `cdktf get`

### Important Factoids

- Running on MBP M1
- Using `asdf-vm` to install node and npx
- `asdf-vm` is aliased to `arch -x86_64 asdf` and all binaries and tools installed by it are x86_64
```
❯ file ~/.asdf/installs/nodejs/14.2.0/bin/node
/Users/USERNAME/.asdf/installs/nodejs/14.2.0/bin/node: Mach-O 64-bit executable x86_64
```

**Comment by jsteinich** [comment]
What version of Terraform do you have installed?

**Comment by pecigonzalo** [comment]
My global version is (arm):

```
❯ terraform -v
Terraform v1.0.3
on darwin_arm64

Your version of Terraform is out of date! The latest version
is 1.0.4. You can update by downloading from https://www.terraform.io/downloads.html
```

My repo version is (with asdf-vm, x86_64):

```
❯ terraform -v
Terraform v0.13.6

Your version of Terraform is out of date! The latest version
is 1.0.4. You can update by downloading from https://www.terraform.io/downloads.html
```

**Comment by ansgarm** [comment]
> So this uses the terraform CLI in the background?

Yes, it does. But you can also supply the `TERRAFORM_BINARY_NAME` environment variable to point the CDKTF to another Terraform binary. ([related source code](https://github.com/hashicorp/terraform-cdk/blob/main/packages/cdktf-cli/bin/cmds/helper/terraform.ts#L4))

**Comment by pecigonzalo** [comment]
🤔 It could be that the exec is not happening in the context of the current folder, and its then grabbing my system `terraform` binary which is `darwin_arm64`.

**Comment by pecigonzalo** [comment]
OK, I have tested changing my global terraform to use an `x86_64` version and then the `cdktf` commands work. 

I think we should still consider it a bug that this does not respect the context from which Im running the command from, and breaks `asdf-vm` version manager and likely other managers that depend on the current folder.
